azure-communication-ui/calling/src/main/java/com/azure/android/communication/ui/calling/presentation/fragment/calling/controlbar/captions/CaptionsListView.kt

```diff
@@ -123,54 +124,69 @@ internal class CaptionsContainerView(
+                    itemType = BottomCellItemType.BottomMenuAction,
+                    onClickAction = null,
+                    showToggleButton = true,
+                    isToggleButtonOn = isCaptionsActive,
+                    isEnabled = canTurnOnCaptions && viewModel.liveCaptionsToggleButton?.isEnabled ?: true,
+                    toggleButtonAction = { _, isChecked ->
+                        try {
+                            viewModel.liveCaptionsToggleButton?.onClickHandler?.handle(
+                                createButtonClickEvent(context, viewModel.liveCaptionsToggleButton)
+                            )
+                        } catch (_: Exception) {
+                        }
+                        viewModel.toggleCaptions(isChecked)
+                    }
+                )
             )
-        )
-        items.add(
-            BottomCellItem(
-                icon = ContextCompat.getDrawable(
-                    context,
-                    R.drawable.azure_communication_ui_calling_ic_fluent_spoken_language_24_selector
-                ),
-                title = context.getString(R.string.azure_communication_ui_calling_captions_spoken_language_title),
-                "",
-                null,
-                null,
-                null,
-                null,
-                null,
-                isOnHold = null,
-                BottomCellItemType.BottomMenuAction,
-                showRightArrow = true,
-                subtitle = LocaleHelper.getLocaleDisplayName(activeSpokenLanguage),
-                onClickAction = {
-                    viewModel.openSpokenLanguageSelection()
-                },
-                isEnabled = isCaptionsActive
+        }
+        if (viewModel.spokenLanguageButtonOptions?.isVisible != false) {
+            items.add(
+                BottomCellItem(
+                    icon = ContextCompat.getDrawable(
+                        context,
+                        R.drawable.azure_communication_ui_calling_ic_fluent_spoken_language_24_selector
+                    ),
+                    title = context.getString(R.string.azure_communication_ui_calling_captions_spoken_language_title),
+                    "",
+                    null,
+                    null,
+                    null,
+                    null,
+                    null,
+                    isOnHold = null,
+                    BottomCellItemType.BottomMenuAction,
+                    showRightArrow = true,
+                    subtitle = LocaleHelper.getLocaleDisplayName(activeSpokenLanguage),
+                    onClickAction = {
+                        try {
+                            viewModel.spokenLanguageButtonOptions?.onClickHandler?.handle(
+                                createButtonClickEvent(context, viewModel.spokenLanguageButtonOptions)
+                            )
+                        } catch (_: Exception) {
+                        }
```

Comment:
same as above, Log exceptions we hide so dev can see them in logcat.

Reply:
Added